solr/core/src/java/org/apache/solr/storage/SizeAwareDirectory.java

```diff
@@ -91,6 +158,26 @@ public long fileLength(String name) throws IOException {
     }
   }
 
+  @Override
+  public long onDiskFileLength(String name) throws IOException {
+    Sizes ret = fileSizeMap.get(name);
+    SizeAccountingIndexOutput live;
+    if (ret != null) {
+      return ret.onDiskSize;
+    } else if ((live = liveOutputs.get(name)) != null) {
+      if (live.backing instanceof CompressingDirectory.SizeReportingIndexOutput) {
+        return ((CompressingDirectory.SizeReportingIndexOutput) live.backing).getBytesWritten();
+      }
+      return live.backing.getFilePointer();
```

Comment:
Pretty sure this is the problem with the original design. `onDiskFileLength()`, according to how we're using it, is supposed to only report data written to _persistent_ disk. Since both Tee IndexOutput (writes to disk) and plain access-only IndexOutput (retrieved via TeeDirectory.createTempOutput()) are tracked via `liveOutputs`, we should fallback to `return 0` here for non-`SizeReportingIndexOutput` -- instead of `return live.backing.getFilePointer()`.

In fact, this is handled correctly elsewhere in this same file -- for [SizeAccountingIndexOutput.getBytesWritten()](https://github.com/cowpaths/fullstory-solr/blob/cc32c888770770f24792a5731996bb3336d3e64a/solr/core/src/java/org/apache/solr/storage/SizeAwareDirectory.java#L461-L466) ... I think we just need to do the same thing here?